insights/parsers/samba.py

```diff
@@ -108,3 +109,31 @@ def parse_content(self, content):
             # Merge same-named sections just as samba's `testparm` does.
             new_dict[new_key].update(old_section)
         self.data._sections = new_dict
+
+
+add_filter(Specs.testparm_v_s, ["["])
+add_filter(Specs.testparm_v_s, ["Server role:"])
+
+
+@parser(Specs.testparm_v_s)
+class SambaConfigs(SambaConfig):
+    """
+    This parser reads the Samba configuration from command `testparm -v -c` which is more
+    reliable than parsing the config as well as contains server role.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.server_role = None
+        super(SambaConfigs, self).__init__(*args, **kwargs)
+
+    def parse_content(self, content):
+        # Parse server role
+        for line in content:
+            r = re.search(r"Server role:\s+(\S+)", line)
+            if r:
+                self.server_role = r.group(1)
```

Comment:
From the above docstring, it seems the `server_role` is a must-exist attribute for this parser. Is it OK to leave its value as `None` (the current logic) when cannot found the `Server role:` line from the spec, OR raise a `ParseException/SkipException` for this case?

Reply:
I looked at the `testparm` source and with the current options in the spec it should be always present. I think parse exception makes sense.